EIPS/eip-3224.md

```diff
@@ -0,0 +1,466 @@
+`eipXXXDescription(bytes)`.
+
+If the wallet contains a user interface for accepting or
+denying a transaction, it should present the description string
+along with fee and value information. Optionally, a wallet may
+wish to additionally provide a way to further examine the
+transaction.
+
+If accepted, the transaction data is set to the computed
+**described data**, the derived transaction is signed (and not
+sent) and the **description string** and serialized signed
+transaction is returned.
+
+### Description Strings
+
+A **description string** must begin with a mime-type followed
+by a semi-colon (`;`). This EIP specifies only the `text/plain`
+mime-type, but future EIPs may specify additional types to
+enable more rich processing, such as `text/markdown` so that
+addresses can be linkable within clients or to enable
+multi-locale options, similar to multipart/form-data.
+
+
+## Rationale
+
+### Meta Description
+
+There have been many attempts to solve this problem, many of
+which attempt to examine the encoded transaction data or
+message data directly.
+
+In many cases, the information that would be necessary for a
+meaningful description is not present in the final encoded
+transaction data or message data.
+
+Instead this EIP uses an indirect description of the data.
+
+For example, the `commit(bytes32)` method of ENS places a
+commitement **hash** on-chain. The hash contains the
+**blinded** name and address; since the name is blinded, the
+encoded data (i.e. the hash) no longer contains the original
+values and is insufficient to access the necessary values to
+be included in a description.
+
+By instead describing the commitment indirectly (with the
+original information intact: NAME, ADDRESS and SECRET) a
+meaningful description can be computed (e.g. "commit to NAME for ADDRESS (with SECRET)")
+and the matching data can be computed (i.e. `commit(hash(name, owner, secret))`).
+
+### Alternatives
+
+- [NatSpec](https://docs.soliditylang.org/en/latest/natspec-format.html) and company; these languages are usually quite large and complex, requiring entire JavaScript VMs with ample processing power and memory, as well as additional sandboxing to reduce security concerns. One goal of this is to reduce the complexity to something that could execute on hardware wallets and other simple wallets.
+
+- Custom Languages; due to the complexity of Ethereum transactions, any language used would require a lot of expressiveness and re-inventing the wheel. The EVM already exists (it may not be ideal), but it is there and can handle everything necessary. 
+
+- Format Strings (e.g. [Trustless Signing UI Protocol](https://github.com/ethereum/EIPs/issues/719)); format strings can only operate on the class of regular languages, which in many cases is insufficient to describe an Ethereum transaction. This was an issue quite often during [early attempts](https://github.com/ricmoo/distinguished-format-strings) at solving this problem
```

Comment:
Generally, we should try to avoid linking to external resources as the links will almost certainly break with time and editors do not have the manpower to maintain them.  It is acceptable to link to published (non-draft) EIPs, but for this section I recommend leaving out the links and specific products and instead just describing the alternative solutions briefly.

Reply:
(same goes for the NatSpec link above)